flow_parser v0.229.1

filipeom opened the pull request:
[flow parser](https://github.com/facebook/flow) v0.229.1

EDIT: Sorry about abusing the CI!

haochenx commented on packages/flow_parser/flow_parser.0.229.1/opam:
```diff
@@ -0,0 +1,28 @@
+build: ["dune" "build" "-p" name "-j" jobs]
+depends: [
+  "ocaml" {>= "4.14.0" & < "5.2"}
+  "dune" {>= "3.2"}
```
in the repository, it seems that only Dune >= 3.0 is required:

- https://github.com/facebook/flow/blob/392b6d3008caa40a38627de876f3cd29611f6be4/flow_parser.opam#L13
- https://github.com/facebook/flow/blob/392b6d3008caa40a38627de876f3cd29611f6be4/dune-project#L1

filipeom replied:
The reason for the difference is the `dune-workspace` in the flow repository uses `3.2` so it was falling some lower bounds CI: https://github.com/facebook/flow/blob/392b6d3008caa40a38627de876f3cd29611f6be4/dune-workspace 
That's why I bumped it to 3.2

haochenx replied:
I see! thanks for the explanation.

(it's probably out of the scope, but IIUC [it is not recommended to check-in dune-workspace](https://dune.readthedocs.io/en/stable/usage.html#:~:text=while%20it%20is%20less%20common%20to%20commit%20dune%2Dworkspace%20files). )

haochenx reviewed (commented):
besides the lower bound for Dune, this PR itself looks good to me

haochenx commented:
It seems that
https://github.com/facebook/flow/blob/v0.229.1/flow_parser.opam
is out-of-date compared to the OPAM file in this PR.

@filipeom it's completely up to you, but it's probably a good idea to upstream the updates.

filipeom commented:
> It seems that https://github.com/facebook/flow/blob/v0.229.1/flow_parser.opam is out-of-date compared to the OPAM file in this PR.
> 
> @filipeom it's completely up to you, but it's probably a good idea to upstream the updates.

I've raised the issue here: https://github.com/facebook/flow/pull/9121. However, it appears that there isn't a significant interest in incorporating these changes.

haochenx commented:
re: https://github.com/ocaml/opam-repository/pull/25339#issuecomment-1965399730

Thank you for the information! It might be a good idea to note this fact as a comment in the opam file IMO.

filipeom commented:
> re: [#25339 (comment)](https://github.com/ocaml/opam-repository/pull/25339#issuecomment-1965399730)
> 
> Thank you for the information! It might be a good idea to note this fact as a comment in the opam file IMO.

Good idea. I will note the changes in a comment so that it doesn't generate confusion

haochenx commented on packages/flow_parser/flow_parser.0.229.1/opam:
```diff
@@ -0,0 +1,29 @@
+opam-version: "2.0"
+maintainer: "flow@fb.com"
```
```suggestion
maintainer: "flow@fb.com"

# note that the opam packages is not officially maintained by the upstream
# see https://github.com/facebook/flow/pull/9121 for upstream discussion
```

haochenx commented:
Many thanks! I have added a suggestion to hint the status of this package to provide more background for other maintainers.

Please check whether it's appropriate.

filipeom commented:
> Many thanks! I have added a suggestion to hint the status of this package to provide more background for other maintainers.
> 
> Please check whether it's appropriate.

Yes, this looks more appropriate then the comments I added. Thanks for all your help :smile:

mseri commented:
Thanks!
